documentation/tutorials/deploy-kernel.rst

```diff
@@ -0,0 +1,179 @@
+
+Now, make sure that you can use kw to login into your target machine::
+
+  kw ssh
+
+If you cannot ssh to the machine, stop here and figure out why you cannot ssh
+to the target machine; after that, get back to the part.
+
+.. note::
+   If it is your first time trying to ssh to the remote machine, try the
+   following command inside it:
+   `sed -i '/#PermitRootLogin without-password/c\PermitRootLogin yes' /etc/ssh/sshd_config && systemctl restart sshd`
+
+If your ssh key has a password, we recommend you to use the below commands
+before you try to deploy your new kernel::
+
+  ssh-agent bash
+  ssh-add ~/.ssh/id_rsa
+
+Since this is the first time that you deploy a new kernel using kw, let's start
+with a command that setup your target machine::
+
+  kw deploy --setup
+
+.. note::
+   All the above configuration represents a one-time setup.
+
+Under the hood, the above command will:
+
+- Make sure that the ssh config is correct.
+- Install required packages in the target machine.
+- Install a small part of kw in the remote.
+
+After the above command completes, you can deploy your change by using the
+command::
+
+ kw deploy # kw d
+
+That's it, kernel installed! You just need to reboot your target system and
+select your new kernel in the GRUB menu.
+
+Local Machine Deploy
+--------------------
+
+In this scenario, a target kernel is your host machine. For example, suppose
+that you want to install the latest stable kernel from the Torvalds tree in
+your laptop; in this case, kw deploy --local is what you are looking for.
+
+Ok, in this case, let's start by entering in your kernel code::
+
+ cd /path/to/kernel
+
+Now, you can use::
+
+ kw deploy --local
+
+.. note::
+   This requires sudo permission, and you will need to select the target kernel
+   after rebooting your system.
+
+If you only deploy your kernel to a local machine, you can change the option
+`default_deploy_target` to `local` in your `.kw/kworflow.config` file.
+
+QEMU VM Deploy
+--------------
+
+Thanks to `guestfish <https://libguestfs.org/guestfish.1.html>`_, kw provides
+close integration with the QEMU VM file. If you want to deploy your new kernel
+to a QEMU VM, you first need to ensure two things:
+
+1. Your VM is off.
+2. Your config file has the correct path to the QEMU image.
+
+If all of the two conditions are true, you just need to run::
+
+  kw deploy --vm
+
+Improving your Workflow
+-----------------------
+
+If you are working on something and you need to build and deploy new kernel
+versions constantly, you can speed up your workflow with these two changes:
+
+1. Open your `.kw/kworflow.config` and set `reboot_after_deploy` with `yes`.
+2. Use `kw bd` which will build your kernel and deploy after that.
+
+In summary with these two changes, your workflow will be:
+
+1. Change something in the kernel code.
+2. Run kw bd.
+3. Validate your change in the target machine.
+4. Go back to step 1 if you are not happy with your change.
+
+Conclusion
```

Comment:
I'd suggest using something like: "Keep in mind", or "Stay stunned" (hehe) to indicate it is in continuous improvement, instead of the feeling of a conclusion. What do you think? I felt it after reading the last paragraph. Also, each kw deploy option has many sub-options..